### PR TITLE
docs: add document for dns chaos

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
Signed-off-by: xiang <xiang13225080@163.com>

add document for dns chaos, should be merged after https://github.com/chaos-mesh/chaos-mesh/pull/1039

### CODE REVIEW

File: docs/chaos_experiments/dns_chaos.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+id: dnschaos_experiment
+title: DNSChaos Experiment
+sidebar_label: DNSChaos Experiment
+---
+
+This document describes how to create DNSChaos experiments in Chaos Mesh.
```
Comment (dcalvin) [inline docs/chaos_experiments/dns_chaos.md:7]: Please explain DNS chaos. What does it do? What's the use scenario

Comment (WangXiangUSTC) [reply docs/chaos_experiments/dns_chaos.md:7]: add in https://github.com/chaos-mesh/website/pull/3/commits/904d01364304c0baaf0e3f625c8a6bdf5a02f075

File: docs/chaos_experiments/dns_chaos.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+* **action** Defines the chaos action about DNS. Supported action: `error` / `random`.
+    - `error` Get error when send DNS request.
+    - `random` Get random IP when send DNS request.
+
+* **scope** Defines the scope in which the DNS chaos works. Supported scope: `outer` / `inner` / `all`.
+    - `outer` DNS chaos only works on the host outer of the Kubernetes cluster.
+    - `inner` DNS chaos only works on the inner host in the Kubernetes cluster.
+    - `all` DNS chaos works on all hosts.
+
+* **selector** specifies the target pods for chaos injection. For more details, see [Define the Scope of Chaos Experiment](../user_guides/experiment_scope.md).
```
Comment (dcalvin) [inline docs/chaos_experiments/dns_chaos.md:61]: - **action**: Defines the chaos action for DNS chaos. Supported actions are:
    - `error` - Get an error when sending the DNS request
    - `random` - Get a random IP when sending the DNS request

* **scope** Defines the scope of the DNS chaos. Supported scopes are:
    - `outer` - DNS chaos only works on outer hosts of the Kubernetes cluster
    - `inner`- DNS chaos only works on inner hosts of the Kubernetes cluster
    - `all` - DNS chaos works on all hosts.

* **selector** specifies the target pods for chaos injection. For more details, see [Define the Scope of Chaos Experiment](../user_guides/experiment_scope.md).

File: docs/chaos_experiments/dns_chaos.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+
+## Notes
+
+- Now DNS chaos only supports record type `A` and `AAAA`.
```
Comment (dcalvin) [inline docs/chaos_experiments/dns_chaos.md:66]:
```suggestion
- Currently, DNSChaos only supports record types `A` and `AAAA`.
```

File: docs/chaos_experiments/dns_chaos.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+- Chaos DNS Service runs coredns with plugin [k8s_dns_chaos](https://github.com/chaos-mesh/k8s_dns_chaos), If the coredns service in your K8s cluster contains some special configurations, you can edit configMap `dns-server-config` to make the configuration of chaos DNS service consistent with the K8s coredns service:
```
Comment (dcalvin) [inline docs/chaos_experiments/dns_chaos.md:67]:
```suggestion
- The Chaos DNS service runs CoreDNS with the  [k8s_dns_chaos](https://github.com/chaos-mesh/k8s_dns_chaos) plugin, If the CoreDNS service in your Kubernetes cluster contains some special configurations, you can edit configMap `dns-server-config` to make the configuration of the chaos DNS service consistent with the K8s CoreDNS service:
```

File: docs/chaos_experiments/dns_chaos.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+sidebar_label: DNSChaos Experiment
+---
+
+This document describes how to create DNSChaos experiments in Chaos Mesh.
+
+DNSChaos allows you to simulate response for DNS requests, include errors or random IP addresses.
+
+## Deploy DNS service for chaos
+
+To create DNSChaos experiments in Chaos Mesh, need to deploy a DNS service in Chaos Mesh by executing the command below:
```
[the remark this answers is not in the document]
Comment (dcalvin) [reply docs/chaos_experiments/dns_chaos.md:13]: Please add the subjective before "need"

**Comment by WangXiangUSTC** [comment]
all comments are addressed, PTAL again @dcalvin

**Comment by WangXiangUSTC** [comment]
@g1eny0ung PTAL

File: docs/chaos_experiments/dns_chaos.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+- **action**: Defines the chaos action for DNS chaos. Supported actions are:
+    - `error` - Get an error when sending the DNS request
+    - `random` - Get a random IP when sending the DNS request
+
+* **scope** Defines the scope of the DNS chaos. Supported scopes are:
+    - `outer` - DNS chaos only works on outer hosts of the Kubernetes cluster
+    - `inner`- DNS chaos only works on inner hosts of the Kubernetes cluster
+    - `all` - DNS chaos works on all hosts.
+
+* **selector** specifies the target pods for chaos injection. For more details, see [Define the Scope of Chaos Experiment](../user_guides/experiment_scope.md).
```
Comment (dcalvin) [inline docs/chaos_experiments/dns_chaos.md:61]:
```suggestion
* **selector** Specifies the target pods for chaos injection. For more details, see [Define the Scope of Chaos Experiment](../user_guides/experiment_scope.md).
```

File: docs/chaos_experiments/dns_chaos.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+
+## Notes
+
+- Currently, DNSChaos only supports record types `A` and `AAAA`.
+- The Chaos DNS service runs CoreDNS with the  [k8s_dns_chaos](https://github.com/chaos-mesh/k8s_dns_chaos) plugin, If the CoreDNS service in your Kubernetes cluster contains some special configurations, you can edit configMap `dns-server-config` to make the configuration of the chaos DNS service consistent with the K8s CoreDNS service:
```
Comment (dcalvin) [inline docs/chaos_experiments/dns_chaos.md:67]:
```suggestion
- The chaos DNS service runs CoreDNS with the  [k8s_dns_chaos](https://github.com/chaos-mesh/k8s_dns_chaos) plugin. If the CoreDNS service in your Kubernetes cluster contains some special configurations, you can edit configMap `dns-server-config` to make the configuration of the chaos DNS service consistent with that of the K8s CoreDNS service as shown below:
```

File: docs/chaos_experiments/dns_chaos.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
+---
+id: dnschaos_experiment
+title: DNSChaos Experiment
+sidebar_label: DNSChaos Experiment
+---
+
+This document describes how to create DNSChaos experiments in Chaos Mesh.
+
+DNSChaos allows you to simulate DNS responses in fault situations including DNS errors or random IP addresses.
```
Comment (dcalvin) [inline docs/chaos_experiments/dns_chaos.md:9]:
```suggestion
DNSChaos allows you to simulate fault DNS responses such as a DNS error or a random IP address after a request is sent.
```

File: docs/chaos_experiments/dns_chaos.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+For more sample files, see [examples](https://github.com/chaos-mesh/chaos-mesh/tree/master/examples). You can edit them as needed.
+
+## Fields description
+
+- **action**: Defines the chaos action for DNS chaos. Supported actions are:
+    - `error` - Get an error when sending the DNS request
+    - `random` - Get a random IP when sending the DNS request
+
+* **scope** Defines the scope of the DNS chaos. Supported scopes are:
```
Comment (dcalvin) [inline docs/chaos_experiments/dns_chaos.md:56]:
```suggestion
* **scope**: Defines the scope of the DNS chaos. Supported scopes are:
```

File: docs/chaos_experiments/dns_chaos.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+    - `outer` - DNS chaos only works on outer hosts of the Kubernetes cluster
+    - `inner`- DNS chaos only works on inner hosts of the Kubernetes cluster
+    - `all` - DNS chaos works on all hosts.
+
+* **selector** Specifies the target pods for chaos injection. For more details, see [Define the Scope of Chaos Experiment](../user_guides/experiment_scope.md).
```
Comment (dcalvin) [inline docs/chaos_experiments/dns_chaos.md:61]:
```suggestion
* **selector**: Specifies the target pods for chaos injection. For more details, see [Define the Scope of Chaos Experiment](../user_guides/experiment_scope.md).
```

**Review by dcalvin — APPROVED** [review]
LGTM

**Pull request merged**